Add coral patches

67protons opened the pull request:
Support for Issue #87 

My first ever commit for a runelite plugin and working in Java, so it's possible I could've done something wrong on accident. Please let me know if I broke anything!

67protons committed:
Add coral patches

67protons commented:
@queicherius Ping on this. Not sure if there's anything I missed on the PR to notify you, or if you're just busy 😅 but would super appreciate this being added to RuneLite!

queicherius commented:
Hey @67protons ! Thank you for the PR :) I just missed this one, because I am busy with the holiday season.

The code itself looks good, but the code in `/runelite/` is generated from the RL source. Could you adjust `update.js` to include your adjustments and run this against the newest RuneLite tag? If that's too annoying I can do that, but it will likely be sometime in the next week.

67protons commented:
Heya, I saw that on the readme, but I wasn't sure what it meant.
Does this mean I need to make a pull request to the official runelite client repo first in order to get those files updated on their end? And then pull down those changes here?

67protons commented:
Oh jk, I think I see it now. There's a string replacement section for the files.

queicherius commented:
> Oh jk, I think I see it now. There's a string replacement section for the files.

Yep its basically a dumb patcher for the RL files :)

67protons committed:
Update update.js

67protons commented:
Alright! I thiiiink I did it correctly? I updated update.js to change `CORAL(Tab.SPECIAL`, checked out runelite tag 1.12.10, and ran update.js. Basically git said I had no changes since I had already updated PatchImplementation.java and Tab.java manually. I tested locally by reverting my changes, running update.js again, and that correctly updated PatchImplementation.

I'm a little confused by Tab.java though. It seems to no longer match the directory from the original runelite repo. In this repo it lives in the `timetracking/farming/` subdir but in the official RL repo it's in the parent `timetracking/` directory - so I'm guessing update.js just doesn't run on that and we have our own local copy here?

queicherius commented:
> Alright! I thiiiink I did it correctly? I updated update.js to change `CORAL(Tab.SPECIAL`, checked out runelite tag 1.12.10, and ran update.js. Basically git said I had no changes since I had already updated PatchImplementation.java and Tab.java manually. I tested locally by reverting my changes, running update.js again, and that correctly updated PatchImplementation.

Awesome, perfect! I validated on my end and it looks all good 💯 

> I'm a little confused by Tab.java though. It seems to no longer match the directory from the original runelite repo. In this repo it lives in the `timetracking/farming/` subdir but in the official RL repo it's in the parent `timetracking/` directory - so I'm guessing update.js just doesn't run on that and we have our own local copy here?

Yes - that one we have our own thing atm. I tend to be happy as long as git is clean after running `update.js`. 🙂
